src/v/archival/archival_policy.cc

```diff
@@ -509,7 +465,11 @@ static ss::future<upload_candidate> create_upload_candidate(
         version = meta->version;
     }
 
-    upload_candidate result{.source = segment};
+    auto deadline = std::chrono::steady_clock::now() + segment_lock_duration;
+    std::vector<ss::rwlock::holder> locks;
+    locks.emplace_back(co_await segment->read_lock(deadline));
```

Comment:
i'm not going to point this out any more in the pr. it looks like there are many instances of this.